app/src/main/java/ch/epfl/polybazaar/MainActivity.java

```diff
@@ -56,24 +49,34 @@ public void clickAuthenticationButton(View view) {
             authenticator.signOut();
             user = authenticator.getCurrentUser();
 
-            updateAuthenticationButton();
+            updateUserButtons();
         }
     }
 
+    public void toProfile(View view){
+        Intent intent = new Intent(MainActivity.this, UserProfileActivity.class);
+        startActivity(intent);
+    }
+
     @Override
     public void onStart() {
         super.onStart();
 
         user = authenticator.getCurrentUser();
-        updateAuthenticationButton();
+        updateUserButtons();
     }
 
-    private void updateAuthenticationButton() {
+    private void updateUserButtons() {
         Button signInBut = findViewById(R.id.authenticationButton);
+        Button profileBut = findViewById(R.id.profileButton);
         if (user != null) {
             signInBut.setText(R.string.sign_out);
+            profileBut.setClickable(true);
+            profileBut.setVisibility(View.VISIBLE);
         } else {
             signInBut.setText(R.string.sign_in);
+            profileBut.setClickable(false);
```

Comment:
I think that `setClickable(false)` is useless because it's done by `setVisibility(View.INVISIBLE)`.

Reply:
But I'm not sure.